src/engines/kotorbase/creatureinfo.cpp

```diff
@@ -0,0 +1,268 @@
+/* xoreos - A reimplementation of BioWare's Aurora engine
+ *
+ * xoreos is the legal property of its developers, whose names
+ * can be found in the AUTHORS file distributed with this source
+ * distribution.
+ *
+ * xoreos is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * xoreos is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with xoreos. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file
+ *  Abstract creature information in KotOR games.
+ */
+
+#include <algorithm>
+
+#include "src/common/util.h"
+
+#include "src/aurora/gff3file.h"
+
+#include "src/engines/kotorbase/creatureinfo.h"
+
+#include "src/engines/kotorbase/gui/chargeninfo.h"
+
+namespace Engines {
+
+namespace KotORBase {
+
+CreatureInfo::ClassLevel::ClassLevel() {
+}
+
+CreatureInfo::ClassLevel::ClassLevel(const ClassLevel &other) {
+	operator=(other);
+}
+
+CreatureInfo::ClassLevel &CreatureInfo::ClassLevel::operator=(const ClassLevel &other) {
+	characterClass = other.characterClass;
+	level = other.level;
+
+	return *this;
+}
+
+CreatureInfo::Skills &CreatureInfo::Skills::operator=(const Skills &other) {
+	computerUse = other.computerUse;
+	demolitions = other.demolitions;
+	stealth = other.stealth;
+	awareness = other.awareness;
+	persuade = other.persuade;
+	repair = other.repair;
+	security = other.security;
+	treatInjury = other.treatInjury;
+
+	return *this;
+}
+
+CreatureInfo::Abilities &CreatureInfo::Abilities::operator=(const Abilities &other) {
+	strength = other.strength;
+	dexterity = other.dexterity;
+	constitution = other.constitution;
+	intelligence = other.intelligence;
+	wisdom = other.wisdom;
+	charisma = other.charisma;
+
+	return *this;
+}
+
+CreatureInfo::CreatureInfo() {
+}
+
+CreatureInfo::CreatureInfo(const Aurora::GFF3Struct &gff) {
+	loadClassLevels(gff);
+	loadSkills(gff);
+	loadAbilities(gff);
+}
+
+CreatureInfo::CreatureInfo(const CharacterGenerationInfo &info) {
+	_levels.resize(1);
+	_levels[0].characterClass = info.getClass();
+	_levels[0].level = 1;
+}
+
+CreatureInfo &CreatureInfo::operator=(const CreatureInfo &other) {
+	_levels = other._levels;
+	_skills = other._skills;
+	_abilities = other._abilities;
+
+	return *this;
+}
+
+int CreatureInfo::getClassLevel(Class charClass) const {
+	auto it = std::find_if(_levels.begin(),
+	                       _levels.end(),
+	                       [&](const ClassLevel &cl) -> bool { return cl.characterClass == charClass; });
```

Comment:
I generally prefer this to be formatted as

```cpp
	auto it = std::find_if(_levels.begin(), _levels.end(), [&](const ClassLevel &cl) {
		return cl.characterClass == charClass;
	});
```

(and the `-> bool` isn't necessary here, since it can be deduced.)